### PR TITLE
(DOCSP-13905): Pass configurations in iOS Tutorial for offline usage 

### PR DESCRIPTION
**Pull request by cbush**
### Jira

- https://jira.mongodb.org/browse/DOCSP-13905

### Staged Changes (Requires MongoDB Corp SSO)

- https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/ios-tut/tutorial/ios-swift/

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

**Comment by cbush** [comment]
Reminder to check that arty-fact works correctly after merge as https://github.com/mongodb/docs-realm/pull/1205 was recently merged and this will test it.

File: source/includes/steps-tutorial-ios-swift-sync.yaml [hunk 1]
```diff
@@ -122,11 +112,15 @@ content: |
...
-  creates ``User`` objects with a Trigger, it can sometimes take a few
-  seconds after account creation for the backend Trigger to generate a
-  user's ``User`` object.
+  Next, open the {+realm+} with the configuration passed in from the previous
+  controller. It's best to pass configurations around rather than {+realm+}.
+  Whether the {+realm+} was opened with Sync or not doesn't really matter to
+  this ViewController.
+
+  Add a change listener at the end of the ``init`` method that watches for a new
+  ``User`` object in the user {+realm+}. Because your backend app creates
```
Comment (dacharyc) [inline source/includes/steps-tutorial-ios-swift-sync.yaml:121]: Do we say anywhere what a `user {+realm+}` is in this context? We're saying "User" a lot in this section and I'm having trouble keeping the state of all of the "user" references consistent in my head. Is it the realm file that this specific user should have access to - i.e. the user's realm? I see that we also say "user Realm" in [the `signIn()` section above](https://github.com/mongodb/docs-realm/pull/1208/files#diff-c1c95ac8e5cb095f67019814e4f391c73274ea6f2ebaebc9ee7790590d402efbR92) - I wonder if we can clarify our meaning in both places.

File: source/includes/steps-tutorial-ios-swift-sync.yaml [hunk 2]
```diff
@@ -162,39 +156,18 @@ content: |
...
-  Update the logic for creating Tasks to set that partition value in
-  ``addButtonDidClick``:
-
-  .. literalinclude:: /tutorial/generated/swift-ios/sync/TasksViewController.codeblock.add-button-did-click.swift
-     :language: swift
-
-  You'll also have to add a new method that checks if a project is owned
-  by the current user, which controls whether or not the user can manage
-  the list of users allowed to access that project:
+  Next, we'll to add menu to manage members of a project that you can open with
```
Comment (dacharyc) [inline source/includes/steps-tutorial-ios-swift-sync.yaml:162]:
```suggestion
  Next, we'll need to add a menu to manage members of a project that you can open with
```
Seems like we're missing some words. But this is also awkwardly worded - perhaps we could say instead something like "Add a button that enables the user to open a menu to manage project members"?

File: source/includes/steps-tutorial-ios-swift-sync.yaml [hunk 3]
```diff
@@ -162,39 +156,18 @@ content: |
...
+  the press of a button. Navigate to TaskViewController.swift. Add a new method
+  that checks if a project is owned by the current user, which controls whether
```
Comment (dacharyc) [inline source/includes/steps-tutorial-ios-swift-sync.yaml:164]: Nit: we could simplify this by making a full stop and new sentence here. i.e. "checks if a project is owned by the current user. This controls whether or not..."

File: source/includes/steps-tutorial-ios-swift-sync.yaml [hunk 4]
```diff
@@ -218,6 +191,13 @@ content: |
   Team view. Since the client side cannot handle access management, we need to
   call out to our Realm functions we defined earlier.
 
+  .. note::
```
Comment (dacharyc) [inline source/includes/steps-tutorial-ios-swift-sync.yaml:194]: Should this be a warning, instead?

**Review by dacharyc — CHANGES_REQUESTED** [review]
LGTM overall. I did find one typo where there were some missing words, and a couple of places where things are worded awkwardly or seemed unclear to me.

I only read the text and code snippets - I didn't actually try cloning and running the tutorial. Would you like me to go through and run the tutorial, too, or is this level of review sufficient?

File: source/includes/steps-tutorial-ios-swift-sync.yaml (hunk 1 above)
Comment (cbush) [reply source/includes/steps-tutorial-ios-swift-sync.yaml:121]: Added a note and also pasted it around to the other tutorials.

File: source/includes/steps-tutorial-ios-swift-local.yaml [hunk 5]
```diff
@@ -179,13 +179,17 @@ content: |
 title: Enable Sign-in
 ref: sign-in
 content: |
-  Finally, implement the ``signIn()`` method to open a unique project
-  based on the username entered in WelcomViewController. Once you've
-  opened the user {+realm+} successfully, navigate to the TasksViewController:
+  Finally, implement the ``signIn()`` method. Create a {+realm+} configuration
```
[the remark this answers is not in the document]
Comment (cbush) [reply source/includes/steps-tutorial-ios-swift-local.yaml:182]: Happy to change, but having trouble here - it feels self-explanatory, so I'm not sure what to say.

**Comment by cbush** [comment]
@dacharyc Thanks for the review! Comments addressed, let me know if you have more thoughts.

File: source/includes/steps-tutorial-ios-swift-local.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/includes/steps-tutorial-ios-swift-local.yaml:182]: Fair. I'm thinking bigger picture about how we talk about configuration relative to opening realms, so I'll make a mental note for myself to put in a link here if we do a "configuration" concept at some point.

File: source/includes/note-tutorial-custom-user-data-permissions.txt [hunk 6]
```diff
@@ -0,0 +1,12 @@
+.. note:: How Do We Know Which Projects a User Can Access?
+   
+   The :ref:`backend you imported <tutorial-task-tracker-create-realm-app>`
+   makes exactly one custom user data object for each user upon signup. The
+   backend is set up so that every user has read-only access to their own
+   custom user data object. The backend also has functions to add and remove
+   access to project, which we will use later when we add the Manage Team
```
Comment (dacharyc) [inline source/includes/note-tutorial-custom-user-data-permissions.txt:7]:
```suggestion
   access to projects, which we will use later when we add the Manage Team
```

File: source/includes/note-tutorial-custom-user-data-permissions.txt [hunk 7]
```diff
@@ -0,0 +1,12 @@
+.. note:: How Do We Know Which Projects a User Can Access?
```
Comment (dacharyc) [inline source/includes/note-tutorial-custom-user-data-permissions.txt:1]: Awesome, this is a really helpful piece of context. Thank you for adding this!

File: source/includes/steps-tutorial-ios-swift-sync.yaml (hunk 1 above)
Comment (dacharyc) [reply source/includes/steps-tutorial-ios-swift-sync.yaml:121]: That note is excellent - provides such good context. Thank you for adding it!

**Review by dacharyc — APPROVED** [review]
Nice, looks great! I found one little thing in the new note, which I've made a suggestion about, but otherwise LGTM.

**Pull request merged**